### PR TITLE
change: add XGBoost support to image_uris.retrieve()

### PR DESCRIPTION
**Pull request by laurenyu**
*Issue #, if available:*
#1464 

*Description of changes:*
I think this is the last of the frameworks with unique edge cases (XGBoost has images both as an Amazon algorithm and as a framework).

*Testing done:*
unit tests, linters

## Merge Checklist

_Put an `x` in the boxes that apply. You can also fill these out after creating the PR. If you're unsure about any of them, don't hesitate to ask. We're here to help! This is simply a reminder of what we are going to look for before merging your pull request._

#### General

- [x] I have read the [CONTRIBUTING](https://github.com/aws/sagemaker-python-sdk/blob/master/CONTRIBUTING.md) doc
- [x] I used the commit message format described in [CONTRIBUTING](https://github.com/aws/sagemaker-python-sdk/blob/master/CONTRIBUTING.md#committing-your-change)
- [ ] I have passed the region in to all S3 and STS clients that I've initialized as part of this change.
- [ ] I have updated any necessary documentation, including [READMEs](https://github.com/aws/sagemaker-python-sdk/blob/master/README.rst) and [API docs](https://github.com/aws/sagemaker-python-sdk/tree/master/doc) (if appropriate)

#### Tests

- [x] I have added tests that prove my fix is effective or that my feature works (if appropriate)
- [ ] I have checked that my tests are not configured for a specific region or account (if appropriate)
- [ ] I have used [`unique_name_from_base`](https://github.com/aws/sagemaker-python-sdk/blob/master/src/sagemaker/utils.py#L77) to create resource names in integ tests (if appropriate)

By submitting this pull request, I confirm that my contribution is made under the terms of the Apache 2.0 license.

### CODE REVIEW

File: tests/unit/sagemaker/image_uris/regions.py [hunk 1]
```diff
@@ -0,0 +1,22 @@
...
+# or in the "license" file accompanying this file. This file is
+# distributed on an "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF
+# ANY KIND, either express or implied. See the License for the specific
+# language governing permissions and limitations under the License.
+from __future__ import absolute_import
+
+import boto3
+
+
+def regions():
```
Comment (metrizable) [inline tests/unit/sagemaker/image_uris/regions.py:18]: i seem to recall seeing this method somewhere else

Comment (laurenyu) [reply tests/unit/sagemaker/image_uris/regions.py:18]: in a recent PR of mine or somewhere else? I refactored the recently added one to this one so that it could be used across files. I did a grep for the botocore region-determination logic, and found only some code that generates EC2 regions in the Neo unit tests
